ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -228,20 +228,23 @@ export default class ConfirmTransactionBase extends Component {
     if (insufficientBalance) {
       return {
         valid: false,
+        hasSimulationError: false,
```

Comment:
Instead of adding `hadSimulationError` to the return of `getErrorKey`, we could just use `const hasSimulationError = Boolean(txData.simulationFails);` on line 321